resources/qml/PrintMonitor.qml

```diff
@@ -20,49 +20,462 @@ Column
-            property string value: printerConnected ? Math.round(connectedPrinter.hotendTemperatures[index]) + "°C" : ""
+            id: extrudersGrid
+            columns: 2
+            columnSpacing: UM.Theme.getSize("sidebar_lining_thin").width
+            rowSpacing: UM.Theme.getSize("sidebar_lining_thin").height
+            width: parent.width
+
+            Repeater
+            {
+                model: machineExtruderCount.properties.value
+                delegate: Rectangle
+                {
+                    id: extruderRectangle
+                    color: UM.Theme.getColor("sidebar")
+                    width: extrudersGrid.width / 2 - UM.Theme.getSize("sidebar_lining_thin").width / 2
+                    height: UM.Theme.getSize("sidebar_extruder_box").height
+                    Layout.fillWidth: index == machineExtruderCount.properties.value - 1 && index % 2 == 0
+
+                    Text //Extruder name.
+                    {
+                        text: machineExtruderCount.properties.value > 1 ? extrudersModel.getItem(index).name : catalog.i18nc("@label", "Hotend")
+                        color: UM.Theme.getColor("text")
+                        anchors.left: parent.left
+                        anchors.leftMargin: UM.Theme.getSize("default_margin").width
+                        anchors.top: parent.top
+                        anchors.topMargin: UM.Theme.getSize("default_margin").height
+                    }
+                    Text //Temperature indication.
+                    {
+                        text: printerConnected ? Math.round(connectedPrinter.hotendTemperatures[index]) + "°C" : ""
+                        font: UM.Theme.getFont("large")
+                        anchors.right: parent.right
+                        anchors.rightMargin: UM.Theme.getSize("default_margin").width
+                        anchors.top: parent.top
+                        anchors.topMargin: UM.Theme.getSize("default_margin").height
+                    }
+                    Rectangle //Material colour indication.
+                    {
+                        id: materialColor
+                        width: materialName.height * 0.75
+                        height: materialName.height * 0.75
+                        color: printerConnected ? connectedPrinter.materialColors[index] : "#00000000" //Need to check for printerConnected or materialColors[index] gives an error.
+                        border.width: UM.Theme.getSize("default_lining").width
+                        border.color: UM.Theme.getColor("lining")
+                        visible: printerConnected
+                        anchors.left: parent.left
+                        anchors.leftMargin: UM.Theme.getSize("default_margin").width
+                        anchors.verticalCenter: materialName.verticalCenter
+                    }
+                    Text //Material name.
+                    {
+                        id: materialName
```

Comment:
The text is an empty string then. It needs to have a fallback for the text anyway or we're accessing `connectedPrinter.materialNames` when `connectedPrinter` is `null`.